shared/provision/paper-key/index.js

```diff
@@ -1,75 +1,74 @@
 // @flow
 import * as React from 'react'
 import * as Constants from '../../constants/provision'
-import {ButtonBar, BackButton, Icon, Box2, Text, PlainInput, WaitingButton} from '../../common-adapters'
-import {
-  globalColors,
-  globalMargins,
-  globalStyles,
-  styleSheetCreate,
-  isMobile,
-  isAndroid,
-  platformStyles,
-} from '../../styles'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
 
 type Props = {
   onBack: () => void,
-  onSubmit: () => void,
-  onChangePaperKey: (val: string) => void,
-  paperKey: string,
+  onSubmit: string => void,
   hint: string,
   error: string,
   waitingForResponse?: ?boolean,
 }
 
-const PaperKey = (props: Props) => (
-  <Box2 direction="vertical" fullWidth={true} fullHeight={true} gap="medium">
-    <BackButton onClick={props.onBack} style={styles.backButton} />
-    <Box2
-      direction="vertical"
-      style={styles.contents}
-      centerChildren={!isAndroid /* android keyboardAvoiding doesnt work well */}
-      gap={isMobile ? 'tiny' : 'medium'}
-    >
-      <Box2 direction="vertical" gap="tiny" centerChildren={true} gapEnd={true}>
-        <Icon type="icon-paper-key-48" />
-        <Text type="Header" style={styles.hint}>
-          {props.hint}
-        </Text>
-      </Box2>
-      <Box2 direction="vertical" style={styles.inputContainer}>
-        <PlainInput
-          autoFocus={true}
-          multiline={true}
-          rowsMax={3}
-          placeholder="Type in your paper key"
-          textType="Header"
-          style={styles.input}
-          onEnterKeyDown={props.onSubmit}
-          onChangeText={props.onChangePaperKey}
-          value={props.paperKey}
-        />
-      </Box2>
-      {!!props.error && <Text type="BodySmallError">{props.error}</Text>}
-      <ButtonBar fullWidth={true}>
-        <WaitingButton
-          label="Continue"
-          type="Primary"
-          fullWidth={true}
-          onClick={props.onSubmit}
-          enabled={!!props.paperKey}
-          waitingKey={Constants.waitingKey}
-        />
-      </ButtonBar>
-    </Box2>
-  </Box2>
-)
+class PaperKey extends React.Component<Props, {paperKey: string}> {
```

Comment:
Might as well be a PureComponent right?

Reply:
I'm not so worried about pure or not on these connected components as the connect hoc is pure